modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServiceImpl.scala

```diff
@@ -86,4 +86,7 @@ class MultilanguageGatewayServiceImpl(aggregateName: String, eventsTopicName: St
 
   }
 
+  override def healthCheck(in: HealthCheckRequest): Future[HealthCheckReply] =
+    bridgeToBusinessApp.healthCheck(in)
```

Comment:
I think here you could use the `surgeEngine.healthCheck` method to check the healthiness of the actual engine. It'll include a bunch of details that you can either include (and expand the protobuf model for the `HealthCheckReply`) or filter out to distill an `UP` or `DOWN` status from the top level of the health check tree.

Reply:
Added

Reply:
@jeffboutotte I agree ...So the healthcheck for this has two conditions..1. is the underlying surge engine healthy (and here we use the surgeEngine.healthCheck) ....and 2. can we connect to the business app via gRPC (and here we use the new gRPC call)